bindata/v3.11.0/openshift-apiserver/deploy.yaml

```diff
@@ -134,12 +134,12 @@ spec:
           periodSeconds: 5
           timeoutSeconds: 10
```

Comment:
I know you're not touching this, but I am curious about the implications on graceful shutdown of having a probe timeout that is greater than the period. If every probe request times out, does the period effectively become 10s?

Reply:
That is a very good question. In the other thread you are suggesting to set `--shutdown-delay-duration` to match the `readinessProbe`. The value for that parameter might be different depending on the proble implementation. Am I right ?

Assuming the period of 5s will be ignored after reaching timeout we should set `--shutdown-delay-duration` to `40s`, am I right?

Reply:
At the moment the shutdown delay is set to `15s`. Increasing this value will delay rollout significantly.

Reply:
Yes, I think so. The selection of 10s for timeoutSeconds appears to have been based on an etcd check (https://issues.redhat.com//browse/OCPBUGS-14010). Do we need to revisit that choice as part of this? Can we make the etcd check asynchronous so that it isn't on the critical path of the apiserver livez and readyz handlers?

Reply:
> Do we need to revisit that choice as part of this?

I don't think so, because we would like to set the etcd timeouts to 9s.

> Can we make the etcd check asynchronous so that it isn't on the critical path of the apiserver livez and readyz handlers

we could, but that would require changing upstream first besides we are going to exclude the etcd checks from the livez

Reply:
After doing some tests, I've found out that if `timeoutSeconds > periodSeconds`, then the kubelet will issue the next probe right after the timeout (if the previous probe times out, that is), and it won't wait for another `periodSeconds`.

This means that 3 readiness checks will fail in the worst case after `5s + 3*10s`.